crates/bevy_text/src/pipeline.rs

```diff
@@ -1,219 +1,382 @@
-        info.min = min;
-        info.max = max;
-        info
+impl TextMeasureInfo {
+    /// Computes the size of the text area within the provided bounds.
+    pub fn compute_size(
+        &mut self,
+        bounds: Vec2,
+        font_system: &mut cosmic_text::FontSystem,
+    ) -> Vec2 {
+        self.buffer
+            .set_size(font_system, Some(bounds.x.ceil()), Some(bounds.y.ceil()));
+        buffer_dimensions(&self.buffer)
     }
+}
 
-    pub fn compute_size(&self, bounds: Vec2) -> Vec2 {
-        let sections = &self.sections;
-        let geom = SectionGeometry {
-            bounds: (bounds.x, bounds.y),
-            ..Default::default()
-        };
-        let section_glyphs = glyph_brush_layout::Layout::default()
-            .h_align(self.justification.into())
-            .line_breaker(self.linebreak_behavior)
-            .calculate_glyphs(&self.fonts, &geom, sections);
-
-        compute_text_bounds(&section_glyphs, |index| {
-            let font = &self.fonts[index];
-            let font_size = self.sections[index].scale;
-            ab_glyph::Font::into_scaled(font, font_size)
-        })
-        .size()
-    }
+/// Translates [`TextSection`] to [`Attrs`](cosmic_text::attrs::Attrs),
+/// loading fonts into the [`Database`](cosmic_text::fontdb::Database) if required.
+fn get_attrs<'a>(
+    section: &'a TextSection,
+    section_index: usize,
+    font_system: &mut cosmic_text::FontSystem,
+    map_handle_to_font_id: &mut HashMap<AssetId<Font>, cosmic_text::fontdb::ID>,
+    fonts: &Assets<Font>,
+    scale_factor: f64,
+) -> Attrs<'a> {
+    let font_handle = section.style.font.clone();
+    let face_id = map_handle_to_font_id
+        .entry(font_handle.id())
+        .or_insert_with(|| {
+            let font = fonts.get(font_handle.id()).expect(
+                "Tried getting a font that was not available, probably due to not being loaded yet",
+            );
+            let data = Arc::clone(&font.data);
+            let ids = font_system
+                .db_mut()
+                .load_font_source(cosmic_text::fontdb::Source::Binary(data));
+            // TODO: it is assumed this is the right font face
+            *ids.last().unwrap()
+
+            // TODO: below may be required if we need to offset by the baseline (TBC)
+            // see https://github.com/pop-os/cosmic-text/issues/123
+            // let font = font_system.get_font(face_id).unwrap();
+            // map_font_id_to_metrics
+            //     .entry(face_id)
+            //     .or_insert_with(|| font.as_swash().metrics(&[]));
+        });
+    let face = font_system.db().face(*face_id).unwrap();
+    // TODO: validate this is the correct string to extract
+    // let family_name = &face.families[0].0;
+    let attrs = Attrs::new()
+        .metadata(section_index)
+        // TODO: this reference, becomes owned by the font system, which is not really wanted...
+        // .family(Family::Name(family_name))
```

Comment:
This is needed for "multiple fonts of different font families in a single `TextBundle`" to work properly.

Investigated in the WG on discord starting here: https://discord.com/channels/691052431525675048/1248074018612051978/1252723090715049984